refactor(common): deprecate isPlatform functions

JeanMeche opened the pull request:
The Web worker platform was removed in v11.
`isPlatformWorkerApp` & `isPlatformWorkerUi` have no purpose since, lets deprecate them!

## PR Type
What kind of change does this PR introduce?

- [x] Refactoring (no functional changes, no api changes)

## Does this PR introduce a breaking change?

- [x] No.

alan-agius4 reviewed (approved):
LGTM, thanks

atscott reviewed (changes requested):
Actually, these functions are public API. If they're exported, they can't be removed with deprecation and then waiting 2 major versions, right?

JeanMeche commented:
Yeah you're right. 
But they don't have any purpose since v11. So I why really wondering about their usage and if the 2 version depreciation period was necessary !

atscott commented:
> So I was really wondering about their usage and if the 2 version depreciation period was necessary !

Unless they were officially already deprecated in v11, yes

JeanMeche committed:
refactor(common): Deprecate useless isPlatform functions

The Web worker platform was removed in v11.
`isPlatformWorkerApp` & `isPlatformWorkerUi` have no purpose since.

JeanMeche commented:
Let's deprecate them then !

pkozlowski-opensource reviewed (approved):
LGTM

Reviewed-for: public-api

AndrewKushnir reviewed (approved):
Reviewed-for: public-api

alxhub commented:
This PR was merged into the repository by commit f49c6d0a2fe2642707205bc0b83def46293cebee.
